src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/GetAzureVM.cs

```diff
@@ -101,14 +101,18 @@ protected override void ExecuteCommand()
         private List<T> GetVMContextList<T>(string serviceName, NSM.DeploymentGetResponse deployment)
             where T : PVM.PersistentVMRoleContext, new()
         {
-            var vmRoles = new List<NSM.Role>(deployment.Roles.Where(
-                r => string.IsNullOrEmpty(Name)
-                  || r.RoleName.Equals(Name, StringComparison.InvariantCultureIgnoreCase)));
+            Func<NSM.Role, bool> typeMatched =
+                r => string.Equals(r.RoleType, PersistentVMRoleStr, StringComparison.OrdinalIgnoreCase);
```

Comment:
Why is it StringComparison.OrdinalIgnoreCase instead of StringComparison.InvariantCultureIgnoreCase ?